object: disallow unsafe OBC fields by default (#15376)
rook/rook: Documentation/Storage-Configuration/Object-Storage-RGW/ceph-object-bucket-claim.md

    * `bucketMaxObjects`: (disabled by default) The maximum number of objects in the bucket as an individual bucket quota. This is useful when the bucket is shared among multiple users.
    * `bucketMaxSize`: (disabled by default) The maximum size of the bucket as an individual bucket quota.
    * `bucketPolicy`: (disabled by default) A raw JSON format string that defines an AWS S3 format the bucket policy. If set, the policy string will override any existing policy set on the bucket and any default bucket policy that the bucket provisioner potentially would have automatically generated.
    * `bucketLifecycle`: (disabled by default) A raw JSON format string that defines an AWS S3 format bucket lifecycle configuration. Note that the rules must be sorted by `ID` in order to be idempotent.
    * `bucketOwner`: (disabled by default)  The name of a pre-existing ceph rgw user account that will own the bucket. A `CephObjectStoreUser` resource may be used to create an ceph rgw user account. If the bucket already exists and is owned by a different user, the bucket will be re-linked to the specified user.

> Several OBC `additionalConfig` fields are disabled by default. Default-disabled additional config

Reviewer: It might be worth mentioning that the risk is more or less equivalent to users creating `CephObjectStoreUser` CRs.

Author: I'm concerned that would downplay the risks admins have to take on by allowing users to run amok. 

I think I would be more inclined to say that the `CephObjectStoreUser` CRs should have their own equivalent warning noting that admins who allow external-namespace support are opening things up to interference.

> fields are dangerous for administrators to allow users control over, and they should be enabled only

Reviewer 2: "dangerous" is a strong word. What about rephrasing as a risk?
```suggestion
fields may be risky for administrators to allow users control over, and they should be enabled only
```
